include/swift/SIL/SILInstruction.h

```diff
@@ -4552,6 +4554,32 @@ class FullApplySite : public ApplySite {
   }
 };
 
+// This is defined out of line to work around the fact that this depends on
+// PartialApplyInst being defined, but PartialApplyInst is a subclass of
+// ApplyInstBase, so we can not place ApplyInstBase after it.
+template <class Impl, class Base>
+SILFunction *ApplyInstBase<Impl, Base, false>::getCalleeFunction() const {
+  SILValue Callee = getCallee();
+
+  while (true) {
+    if (auto *FRI = dyn_cast<FunctionRefInst>(Callee)) {
+      return FRI->getReferencedFunction();
+    }
+
+    if (auto *PAI = dyn_cast<PartialApplyInst>(Callee)) {
+      Callee = PAI->getCallee();
+      continue;
+    }
+
+    if (auto *TTTFI = dyn_cast<ThinToThickFunctionInst>(Callee)) {
+      Callee = TTTFI->getCallee();
+      continue;
+    }
+
+    return nullptr;
+  }
+}
+
```

Comment:
> On Feb 22, 2016, at 6:37 PM, eeckstein <notifications@github.com> wrote:
> 
> In include/swift/SIL/SILInstruction.h <https://github.com/apple/swift/pull/1393#discussion_r53729431>:
> 
> > +    }
> > +
> > +    if (auto *PAI = dyn_cast<PartialApplyInst>(Callee)) {
> > +      Callee = PAI->getCallee();
> > +      continue;
> > +    }
> > +
> > +    if (auto *TTTFI = dyn_cast<ThinToThickFunctionInst>(Callee)) {
> > +      Callee = TTTFI->getCallee();
> > +      continue;
> > +    }
> > +
> > +    return nullptr;
> > +  }
> > +}
> > +
> Why don't you just add a new function with this more general implementation?
> 
> 

That was my intention once I ran into the error this afternoon, and you confirmed the issue on the list. I would rather not introduce bugs here, so it makes sense to keep all uses besides the getMemoryBehavior() one to use the old code path. Then people can later reason about these code paths and see if it is or is not appropriate.
> —
> Reply to this email directly or view it on GitHub <https://github.com/apple/swift/pull/1393/files#r53729431>.
>